cassiopeia/solver/SpectralNeighborJoiningSolver.py

```diff
@@ -0,0 +1,318 @@
+"""
+This file stores a subclass of DistanceSolver, SpectralNeighborJoining. This 
+algorithm is based on the one developed by Jaffe, et al. (2021) in their paper 
+titled "Spectral neighbor joining for reconstruction of latent tree models",
+published in the SIAM Journal for Math and Data Science. 
+ 
+"""
+
+import itertools
+import networkx as nx
+import numpy as np
+import pandas as pd
+from scipy.linalg import svd
+from typing import Callable, Dict, List, Optional, Tuple
+
+from cassiopeia.solver.DistanceSolver import DistanceSolver, DistanceSolverError
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.solver import (
+    dissimilarity_functions,
+    solver_utilities,
+)
+
+
+class SpectralNeighborJoiningSolver(DistanceSolver):
+    """
+    Spectral Neighbor-Joining class for Cassiopeia.
+
+    Implements a variation on the Spectral Neighbor-Joining
+    algorithm described by Jaffe et al. in their 2020 paper. This class
+    implements the abstract class DistanceSolver. This class inherits
+    the 'solve' method from DistanceSolver with the 'find_cherry' method
+    implemented using a spectral method as in Jaffe et al.  (2020).
```

Comment:
Let's describe the method a bit more. How does Jaffe et al 2020 propose you find cherries? Describe briefly the algorithm - including what the Lambda matrix and how you find cherries.

Reply:
^ This still should be done.